src/argilla/client/feedback/schemas/records.py

```diff
@@ -232,40 +263,126 @@ class RemoteFeedbackRecord(FeedbackRecord):
 
     Args:
-        id: The ID of the record in Argilla. Defaults to None, and is automatically
-            fulfilled internally once the record is pushed to Argilla.
         client: The Argilla client to use to push the record to Argilla. Is shared with
             the `RemoteFeedbackDataset` that created this record.
         name2id: A dictionary that maps the question names to their corresponding IDs.
+        id: The ID of the record in Argilla. Defaults to None, and is automatically
+            fulfilled internally once the record is pushed to Argilla.
+        suggestions: A list of `RemoteSuggestionSchema` that contains the suggestions
+            for the current record in Argilla. Every suggestion is linked to only one
+            question. Defaults to an empty list.
     """
 
     client: httpx.Client
     name2id: Dict[str, UUID]
 
     id: UUID
+    suggestions: Union[Tuple[RemoteSuggestionSchema], List[RemoteSuggestionSchema]] = Field(
+        default_factory=tuple, allow_mutation=False
+    )
+
+    # TODO(alvarobartt): here to be able to use the `allowed_for_roles` decorator
+    @property
+    def _client(self) -> httpx.Client:
+        return self.client
 
     @allowed_for_roles(roles=[UserRole.owner, UserRole.admin])
     def update(
-        self, suggestions: Union[SuggestionSchema, List[SuggestionSchema], Dict[str, Any], List[Dict[str, Any]]]
+        self,
+        suggestions: Union[
+            RemoteSuggestionSchema,
+            List[RemoteSuggestionSchema],
+            SuggestionSchema,
+            List[SuggestionSchema],
+            Dict[str, Any],
+            List[Dict[str, Any]],
+        ],
     ) -> None:
         """Update a `RemoteFeedbackRecord`. Currently just `suggestions` are supported.
 
         Note that this method will update the record in Argilla directly.
 
         Args:
-            suggestions: can be a single `SuggestionSchema`, a list of `SuggestionSchema`,
-                a single dictionary, or a list of dictionaries. If a dictionary is provided,
-                it will be converted to a `SuggestionSchema` internally.
+            suggestions: can be a single `RemoteSuggestionSchema` or `SuggestionSchema`,
+                a list of `RemoteSuggestionSchema` or `SuggestionSchema`, a single
+                dictionary, or a list of dictionaries. If a dictionary is provided,
+                it will be converted to a `RemoteSuggestionSchema` internally.
 
         Raises:
             PermissionError: if the user does not have either `owner` or `admin` role.
         """
-        super().update(suggestions)
-        for suggestion in self.suggestions:
-            suggestion.question_id = self.name2id[suggestion.question_name]
-            datasets_api_v1.set_suggestion(
-                client=self.client, record_id=self.id, **suggestion.dict(exclude_none=True, exclude={"question_name"})
+        if isinstance(suggestions, (dict, SuggestionSchema, RemoteSuggestionSchema)):
```

Comment:
Since `RemoteSuggestionSchema` inherits from `SuggestionSchema`, think we can keep only the base one.
```suggestion
        if isinstance(suggestions, (dict, SuggestionSchema)):
```